src/group_theory/group_action/subgroup.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import group_theory.group_action.basic
+import data.fintype.basic
+/-!
+# Conjugation action on subgroups
+
+This file defines the conjugation action of a group on its subgroups.
+-/
+
+open mul_action
+
+variables {G : Type*} [group G]
+
+/-- The conjugation action of `G` on the subgroups of `G` -/
+instance subgroup.mul_action' : mul_action G (subgroup G) :=
+{ smul := λ g H, H.map (mul_aut.conj g).to_monoid_hom,
+  one_smul := λ _, by ext; simp,
+  mul_smul := λ _ _ _, by ext; simp }
```

Comment:
Having seen [this thread]([this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Lost.20instance/near/249468775)), my impression is that we should:

* Define a new typeclass `mul_monoid_action` (or a better name) that changes the hierarchy from
  ```
  mul_action <- distrib_mul_action <- mul_semiring_action
  ```
  to
  ```
  mul_action <- distrib_mul_action <-\
      ^                              |
      \-- mul_monoid_action <- mul_semiring_action
  ```
  where `mul_monoid_action` contains the `smul_one` and `smul_mul` currently in `to_distrib_mul_action`
* Show that the conjugation action is a `mul_monoid_action`
* Show that any `mul_monoid_action` on `G` lifts to a `mul_monoid_action` on subgroups of `G`.

Reply:
I PR'd this new typeclass in #8918

Reply:
A shorter spelling is now:
```suggestion
open_locale pointwise

instance subgroup.mul_action' : mul_action G (subgroup G) :=
mul_action.comp_hom _ mul_aut.conj
```